docs/design-notes/protocols.md

```diff
@@ -93,19 +93,56 @@ protocol. Scylla supports several Scylla-only extensions to this protocol,
 described in [protocol-extensions.md](protocol-extensions.md).
 
 By default, Scylla listens to the CQL protocol on port 9042, which can be
-configured via the `native_transport_port` configuration option.
+configured via the `native_transport_port` configuration option. If set
+to 0, the OS will choose the port randomly (which also applies to every CQL
+port discussed later). To explicitly disable listening on a CQL port one
+should set its value to `~` , or `null`.
 
 Scylla also supports the CQL protocol via TLS/SSL encryption which is
 disabled by default and can be enabled via the `native_transport_port_ssl`
-configuration option (the traditional choice is port 9142). Users
-that want to only support the encrypted CQL protocol can disable the
-unencrypted default CQL protocol by setting the `native_transport_port` to 0.
+configuration option. Traditionally, the secure connections are accepted
+on port 9142, but if `client_encryption_options` are specified and
+`native_transport_port_ssl` is not, then `native_transport_port` will
+handle encrypted connections only. The same thing happens when
+`native_transport_port` and `native_transport_port_ssl` are set to the same
+value. The rules governing port assignment/encryption are summed up in
+the table below:
+
+```
+np  := native_transport_port is set
+nps := native_transport_port_ssl is set
+ceo := client_encryption_options are enabled
+eq  := native_transport_port_ssl == native_transport_port
+
++-----+-----+-----+-----+
+|  np | nps | ceo |  eq |
++-----+-----+-----+-----+
+|  0  |  0  |  0  |  *  |   =>   listen on native_transport_port, unencrypted
+|  0  |  0  |  1  |  *  |   =>   listen on native_transport_port, encrypted
+|  0  |  1  |  0  |  *  |   =>   don't listen
+|  0  |  1  |  1  |  *  |   =>   listen on native_transport_port_ssl, encrypted
+|  1  |  0  |  0  |  *  |   =>   listen on native_transport_port, unencrypted
+|  1  |  0  |  1  |  *  |   =>   listen on native_transport_port, encrypted
+|  1  |  1  |  0  |  *  |   =>   listen on native_transport_port, unencrypted
+|  1  |  1  |  1  |  0  |   =>   listen on native_transport_port, unencrypted + native_transport_port_ssl, encrypted
+|  1  |  1  |  1  |  1  |   =>   listen on native_transport_port(_ssl - same thing), encrypted
++-----+-----+-----+-----+
+```
+
+To allow "advanced shard-awareness" Scylla accepts CQL connections on
+additional port(s): `native_shard_aware_transport_port` (by default 19042)
+and `native_shard_aware_transport_port_ssl` (traditionally 19142). They work
```

Comment:
There is no tradition for shard aware ports ;-)
Did you try to say that this is the recommended port, but it's not turned on by default? Or *is* it turned on by default somehow with this number?

Reply:
Exactly, this is the recommended port, but it's not turned on by default. I'll rephrase it.

Reply:
Rephrased